### DISCUSSION TITLE
Should I be using a reactive variable in my list-detail view, and if so where?

### DISCUSSION THREAD

**Discussion by lofidevops** [Textualize/textual#2591] [Q&A] [answered]
Below is a simple list-detail layout for a ticket browser, intended as a minimal front-end to GitHub issues, Jira or the like. A wrapper object is used to query the backend. When you pick a list item, the detail pane is reset. This is a familiar pre-AJAX MVC pattern. My intuition is that I "should" rather define a reactive variable, especially if I can make the code less verbose, but I couldn't figure out where. Perhaps the `ticket_id`? If so, how do I trigger an update to the detail pane when the `ticket_id` is changed?

```
# Copyright 2023 David Seaward
# SPDX-License-Identifier: CC0-1.0

SOURCE: SourceWrapper
empty_options = [("", "")]

class TicketBody(VerticalScroll):

    def compose(self):
        global SOURCE

        yield Label("Ticket", id="TicketLabel")
        yield Label("...", id="TicketText")
        yield Label("Sprint", id="SprintLabel")
        yield Select(empty_options, id="SprintSelect")
        yield Label("Epic", id="EpicLabel")
        yield Select(empty_options, id="EpicSelect")
        yield Label("Description", id="DescriptionLabel")
        yield Label("...", id="DescriptionText")

    def reset_values(self, ticket_id):
        global SOURCE
        ticket = SOURCE.get_ticket(ticket_id)

        ticket_text = self.query_one("#TicketText", Label)
        ticket_text.update(ticket.full_title)
        sprint_select = self.query_one("#SprintSelect", Select)
        sprint_select.set_options(SOURCE.get_sprint_select_items())
        sprint_select.value = ticket.sprint_reference
        epic_select = self.query_one("#EpicSelect", Select)
        epic_select.set_options(SOURCE.get_epic_select_items())
        epic_select.value = ticket.epic_reference
        description_text = self.query_one("#DescriptionText", Label)
        description_text.update(ticket.description)


class TicketBrowser(App):
    CSS_PATH = "browser.style"

    def compose(self) -> ComposeResult:
        global SOURCE

        yield OptionList(*SOURCE.get_ticket_option_items(), id="sidebar")
        yield TicketBody(id="body")
        yield Footer()

    @on(OptionList.OptionHighlighted)
    def on_option_list_option_highlighted(self, event: OptionList.OptionHighlighted) -> None:
        if event.option_id is not None:
            body = self.query_one("#body", TicketBody)
            body.reset_values(event.option_id)
```

**Comment by TomJGooding** [comment, marked as answer]
> My intuition is that I "should" rather define a reactive variable, especially if I can make the code less verbose [...] If so, how do I trigger an update to the detail pane

Textual has [watch methods ](https://textual.textualize.io/guide/reactivity/#watch-methods) that will be called when when reactive attributes are modified. But you would still need to handle all the updates to your 'detail pane' - so unless I am missing something I don't think adding a reactive attribute would add much benefit here, at least from your example code?

(I would suggest getting rid of all those `global` variables though!)
